### ISSUE TITLE
`AnimatedTabBarView` proposal

### ISSUE THREAD

**Issue by FMorschel** [flutter/flutter#123888]
## Proposal

My proposal is for an `AnimatedTabBarView`.

This would be similar to an `AnimatedListView` in a way, where we would be given an animation and use it as we wanted. I'm proposing this as a new Flutter Widget because it would only animate the current page as it leaves and the new page as it appears. I've tried to create this but when using something like a `TabBar`, it would not work as well as I wished because it would keep animating as if it was passing through all the in-between tabs even though they would not appear on the screen for the animation.

Since it's already animating the tabs internally, I figured it would be somewhat easy to implement this.

**Comment by darshankawar** [comment]
Thanks for the proposal @FMorschel 
Have you already looked at pub.dev to see if there's already a community package that offers this feature ?
I see quite a few packages https://pub.dev/packages?q=animated+tabbar+ which you can check and see if they serve your purpose.

**Comment by FMorschel** [comment]
Yes, I've looked and none of them does what I've said. Mostly all of those, only give different `TabBar` widgets, not `TabBarView`. My proposal is so that when it's animating between views it would also give an animation object just like `AnimatedLisView` so that from going left/right we could also make the view resize or translate or anything similar.

**Comment by FMorschel** [comment]
No, it's not what I'm proposing. This old issue you mentioned is a discrepancy between the animation time of the `TabBarView` and the `TabBar`. I propose that when the `TabBarView` animates between the different tabs, it could have a new parameter that gives us something like `Widget Function(BuildContext context, Animation<double> animation, Widget child)` so that we could make the transitions between the tabs have custom animations.

**Comment by darshankawar** [comment]
Thanks for the update.

**Comment by HansMuller** [comment]
This is an interesting proposal. We don't have time to work on it at the moment but would welcome a PR that covered all the tricky cases.
